### PR TITLE
dependency: update go version to 1.21.6

### PR DESCRIPTION
**Pull request by ivanvc**
Upgrades go version to 1.21.6.

Part of etcd-io/etcd#17269.

### CODE REVIEW

**Review by jmhbnz — APPROVED** [review]
LGTM on workflow green - Thanks @ivanvc

File: cmd/bbolt/README.md [hunk 1]
```diff
@@ -72,7 +72,7 @@
   ```bash
   $bbolt version
   bbolt version: 1.3.7
-  Go Version: go1.21.5
+  Go Version: go1.21.6
```
Comment (ahrtr) [inline cmd/bbolt/README.md:75]: minor comment: It's just an example, we don't need to update it each time when we bump go version.

**Review by ahrtr — APPROVED** [review]
lgtm with a minor comment

Thanks.

**Pull request merged**